[DNM] Feast changes

Vikash082 opened the pull request.

Vikash082 committed:
adding feast submodule

Vikash082 committed:
added custom provider

Vikash082 committed:
adding patch file

Vikash082 committed:
changed patch format

Vikash082 committed:
Added install instructions in README

Vikash082 committed:
Changes to delegate online store processing to remote server

- changes to proxy materialize call to remote online server.
- refactored online store with provision of driver.

Vikash082 committed:
Modified online store to load both local & remote driver

- Added .env file to set Online store config.

Vikash082 committed:
Added requirements.txt for client SDK

Vikash082 committed:
Registry config cleanup

(cherry picked from commit a7f263016cacfe4efd4fb62534777e6265f41a53)

Vikash082 committed:
Minor changes in mysqlserver query

Vikash082 committed:
Module renaming of provider

- The module name of provider, offline store & online store is
  changed.
- The .env file is now expected to be in HOME directory with
  all the config details.

Vikash082 committed:
mysqlserver connect_args bug fix

Vikash082 committed:
Refactored "dkube" folder to "dkubefs"

Vikash082 committed:
dkube sdk addition

Vikash082 committed:
utility changes, registry changes, dkube client changes

Vikash082 committed:
Modified materialize api client to add "project"

- in materialize api, client now send project name also.

Vikash082 committed:
Added instructions in README

Vikash082 committed:
env variable name changes according to Dkube

Other changes:
- added "Host" in request header for online server
- instanceof() fix
- changed DKUBE_URL

Vikash082 committed:
removed the redundant arg in mysqlserver driver

Vikash082 committed:
mysql 'auth_plugin' changes in connect_args

Vikash082 committed:
removed DKUBE_DATASET environment variable

Vikash082 committed:
Changed the 'mysql-connector' with 'mysql-connector-python'

Vikash082 committed:
Added depepndency packages of python3.9 for feast

Vikash082 committed:
dockerfile and support script to prepare feast ds image

Vikash082 committed:
Merge pull request #1 from oneconvergence/feast_changes_dkrfile

dockerfile and support script to prepare feast ds image

Vikash082 committed:
Changes to support username and offline-dataset in client

Vikash082 committed:
Merge branch 'feast_changes' of https://github.com/oneconvergence/feast into feast_changes

songole commented on dockerfiles/Dockerfile:
```diff
@@ -0,0 +1,93 @@
+
+
+RUN cd /opt \
+    && git clone https://github.com/oneconvergence/feast.git \
```
Just a nit: Would --recurse-submodules clone everything in one go.

songole commented on dockerfiles/Dockerfile:
```diff
@@ -0,0 +1,93 @@
+    apt-get -y autoremove && \
+    rm -rf /var/lib/apt/lists/*
+
+COPY install_python.sh install_python.sh
```
is this script not in the repo?

songole commented on provider/sdk/dkubefs/dkube_client.py:
```diff
@@ -0,0 +1,115 @@
+    def __init__(self, **kwargs) -> None:
+        self.dkube_ip = kwargs.get("dkube_ip", "192.168.x.y")
+        self.dkube_port = kwargs.get("dkube_port", 32222)
+        self.dkube_endpoint = kwargs.get("dkube_endpoint", True)
```
Can we just take dkube_url and token? Can we also get from env right here as opposed to in some caller function?

songole commented on provider/sdk/dkubefs/dkube_provider.py:
```diff
@@ -0,0 +1,7 @@
+
+class DkubeProvider(PassthroughProvider):
+    """
+    (VK): We need to see if we can work with the PassthroughProvider
```
Is the comment still valid?

songole commented on provider/sdk/dkubefs/dkube_client.py:
```diff
@@ -0,0 +1,115 @@
+                verify=verify,
+                params=params,
+                timeout=timeout)
+        return self.process_response(resp)
```
If this is file is mostly independent of the rest of feast, can it be moved to dkube repo?

songole commented on provider/sdk/dkubefs/dkube_registry.py:
```diff
@@ -0,0 +1,53 @@
+
+    def update_registry_proto(self, registry_proto: RegistryProto, **kwargs):
+        registry_proto.version_id = str(uuid.uuid4())
+        registry_proto.last_updated.FromDatetime(datetime.utcnow())
```
can dkube controller do lines 36 & 37, when you update a table?

songole commented on provider/sdk/dkubefs/dkube_registry.py:
```diff
@@ -0,0 +1,53 @@
+    def teardown(self, **kwargs):
+        self.dkube.delete_registry(kwargs['project'])
+
+    def validate_infra_update_with_registry(self, update_infra):
```
What is infra_update call used for?

Vikash082 replied:
This is called before commiting the changes to registry as part of apply call.

songole commented on provider/sdk/dkubefs/dkube_store.py:
```diff
@@ -0,0 +1,114 @@
+        self,
+        config: RepoConfig,
+        tables_to_delete: Sequence[Union[FeatureTable, FeatureView]],
+        tables_to_keep: Sequence[Union[FeatureTable, FeatureView]],
```
You mean update as opposed to keep?

If a table is updated, does it appear in both delete and keep?

Vikash082 replied:
If there is new FV definition, the older one with registry will get deleted and the new definition will get stored. If the table is updated it will be part of table_to_keep.

songole commented on provider/sdk/dkubefs/dkube_store.py:
```diff
@@ -0,0 +1,114 @@
+        # Replace the code below in order to define your own custom teardown
+        # operations
+        self.initialize(config)
+        self.driver["remote"].teardown(config, tables, entities)
```
Are we not tearing down all the project related resources? If so, we just need project?

Vikash082 replied:
This is online store teardown. This is not entire project teardown.

songole replied:
Is Online store not at project level? can we not figure out the rest if we just know project? 
Or, is that how they defined the API?

Vikash082 replied:
This is how the APIs are defined. We need to extend this and take action based on that.

songole commented on provider/sdk/dkubefs/mysqlserver.py:
```diff
@@ -0,0 +1,520 @@
+from dataclasses import dataclass
```
Is this offline source? Why would this be different from what feast supports?

Or, does feast support mysql offline source? If it doesn't, we can propose this to the community?

songole commented on provider/sdk/dkubefs/online_drivers/local_driver.py:
```diff
@@ -0,0 +1,222 @@
+        port = self.dkube_store["port"]
+        user = self.dkube_store["user"]
+        password = self.dkube_store["secret"]
+        database = self.dkube_store["db"]
```
Basically, this supports an external mysql server (like dbaas) as online server, not the one dkube launches. Is this true?

Vikash082 replied:
Yes, this driver is to connect any MYSQL server.

songole commented on feast.patch:
```diff
@@ -0,0 +1,552 @@
+         """
++        if "dkuberegistrystore" == self._registry._registry_store.__class__.__name__.lower():
++            # Proxy materialize call to online server.
++            self._get_provider().proxy_materialize_incremental(
```
proxy_xxx() <-- Are these specific calls we added?

songole commented on feast.patch:
```diff
@@ -0,0 +1,552 @@
+         )
+         return job
++
++    def proxy_materialize(
```
Are we using PassThrough provider?

songole commented on feast.patch:
```diff
@@ -0,0 +1,552 @@
++        }
++        for fs in services_to_update]
++
++        infra_update = {
```
In feast main branch, they have a way to calculate diff. Is it something new? or, just refactored?

songole replied:
https://github.com/feast-dev/feast/tree/master/sdk/python/feast/diff

songole commented on provider/sdk/dkubefs/online_drivers/remote_driver.py:
```diff
@@ -0,0 +1,279 @@
+            "user": self.user,
+            "offline_dataset": self.offline_dataset
+        }
+        self.online_server_client.delete("api/v1/teardown",
```
Can we call dkube sdk whenever we need to make API calls?

Vikash082 replied:
This call goes for knative service independent of controller. This is the reason, not part of Dkube SDK.

songole commented on provider/sdk/dkubefs/utils.py:
```diff
@@ -0,0 +1,173 @@
+        dclient = get_dkube_client()
+        ods = dclient.get_dataset(USER, offline_ds)
+        return {
+            "user": ods["datum"]["sql"]["username"],
```
Can we discuss this?

Vikash082 committed:
Removed IP, port as args from dkubeclient

Vikash082 committed:
Added doc string & updated README

Vikash082 committed:
removed python-decouple module

Vikash082 committed:
bumping feast to v0.22-branch

Vikash082 committed:
Added Git based feast project rendering

Vikash082 committed:
latest feast patch

Vikash082 committed:
update dkube project at time of project creation

Vikash082 committed:
Changes for git based feast init and other changes

- changes in registry to send the changed resource per object basis.
- changes of 'event_timestamp_column' to 'timestamp_field'

Vikash082 committed:
datasource event_timestamp column field fix and minor fixes

- BearerToken tested
- feature store fix where feature store is sending duplicate values.

Vikash082 committed:
changed the provider manipulation of table name

Vikash082 committed:
updated patch

Vikash082 committed:
feast sdk image fix
